### ISSUE TITLE
Finetune  (german/🇩🇪) Thorsten-Voice model | mixed results after 40k steps

### ISSUE THREAD

**Issue by thorstenMueller** [canopyai/Orpheus-TTS-0.1#273]
Guude 👋, i finetune an OrpheusTTS model based on my neutral german Thorsten-Voice dataset (2022.10) using unsloth notebook. The dataset contains around 12000 wave recordings in 24kHz sample rate.
After reaching 40k training steps and synthesizing some test audio i get mixed results.

Training loss started around 5 and went down to 0.5 during training process. I activated wandb logging recently at step 37k.

<img width="2120" height="1350" alt="Image" src="https://github.com/user-attachments/assets/bc846730-e042-4793-9082-ea9bf8d909c4" />

Here's the training config block from jupyter notebook (unsloth):
```
from transformers import TrainingArguments,Trainer,DataCollatorForSeq2Seq

trainer = Trainer(
    model = model,
    train_dataset = dataset,
    args = TrainingArguments(
        per_device_train_batch_size = 1,
        gradient_accumulation_steps = 4,
        warmup_steps = 5,
        num_train_epochs = 7, # Starte mit 7 Epochen (ergibt 7 * 3000 = 21000 Schritte)
        max_steps = 70000, # WICHTIG: Dadurch wird num_train_epochs aktiv.
        learning_rate = 2e-4,
        logging_steps = 100, # Weniger Log-Ausgaben für bessere Übersicht
        optim = "adamw_8bit",
        weight_decay = 0.01,
        lr_scheduler_type = "linear",
        seed = 3407,
        output_dir = "/content/drive/MyDrive/OrpheusTTS/output_august2025/",
        report_to = "wandb", # Ändere dies, wenn du z.B. WandB nutzen möchtest
        # Fügen Optional weitere Argumente hinzu:
        save_steps = 1000, # Speichere Checkpoints alle 5000 Schritte
        # save_total_limit = 2, # Halte nur die letzten 2 Checkpoints
        # evaluation_strategy = "epoch", # Wenn du ein validation_dataset hast
    ),
)
```

And here are some audio samples:
**38k Checkpoint**
*"Ich frage mich wirklich, wieso einige Beispiele sehr gut und andere wiederum überhaupt nicht funktionieren. Wie kann das nur sein?"*
🔈 https://github.com/user-attachments/assets/f88e9bf9-68ee-4e37-9259-62340b562fc8

*"Dieses Beispiel wurde mit der künstlichen, kostenlosen und ohne Internet funktionierenden Thorsten Stimme erzeugt."*
🔈 https://github.com/user-attachments/assets/e444d6ba-1925-4f38-ad82-29a807d06f96

*"Für mich sind alle Menschen gleich, unabhängig von Geschlecht, sexueller Orientierung, Religion, Hautfarbe oder Geokoordination der Geburt."* (i know about the funny typo 🤣)
🔈 https://github.com/user-attachments/assets/90acaf83-942c-4435-b136-b658f2acb5f1

**46k Checkpoint**
*"Ich frage mich wirklich, wieso einige Beispiele sehr gut und andere wiederum überhaupt nicht funktionieren. Wie kann das nur sein?"*
🔈 https://github.com/user-attachments/assets/4242125f-c650-405e-a16d-3b00cfe50736

*"Dieses Beispiel wurde mit der künstlichen, kostenlosen und ohne Internet funktionierenden Thorsten Stimme erzeugt."*
🔈 https://github.com/user-attachments/assets/d65f51e8-1cad-4103-a80b-1d312077fc11

*"Irgendwie traue ich dem Frieden nicht, das wirkt manchmal wirklich sehr zufällig in Punkto Qualität."*
🔈 https://github.com/user-attachments/assets/95e62214-a5a3-47cc-93b8-b6a4bf2f4a07

Do you have any suggestions?
Is more training required or might it be some struggle on missing phonemes in training data?

Best
Thorsten

**Comment by amuvarma13** [comment]
> Guude 👋, i finetune an OrpheusTTS model based on my neutral german Thorsten-Voice dataset (2022.10) using unsloth notebook. The dataset contains around 12000 wave recordings in 24kHz sample rate. After reaching 40k training steps and synthesizing some test audio i get mixed results.
> 
> Training loss started around 5 and went down to 0.5 during training process. I activated wandb logging recently at step 37k.
> 
> <img alt="Image" width="2000" height="1350" src="https://private-user-images.githubusercontent.com/9558265/477737749-bc846730-e042-4793-9082-ea9bf8d909c4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxMjYyOTgsIm5iZiI6MTc1NTEyNTk5OCwicGF0aCI6Ii85NTU4MjY1LzQ3NzczNzc0OS1iYzg0NjczMC1lMDQyLTQ3OTMtOTA4Mi1lYTliZjhkOTA5YzQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MTNUMjI1OTU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGRjOGRjODc0N2RkYTQ5NTNlNGQzM2JkMTEwZDQ3NGFhYWVhMzFmYzE0OGI2YzU3YWNiMzE5MmI3YjE5MGQ0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.k0pSwGxjlIYRfhQtaa6C3iRtaWU4iqx6in6357v1-vE">
> Here's the training config block from jupyter notebook (unsloth):
> 
> ```
> from transformers import TrainingArguments,Trainer,DataCollatorForSeq2Seq
> 
> trainer = Trainer(
>     model = model,
>     train_dataset = dataset,
>     args = TrainingArguments(
>         per_device_train_batch_size = 1,
>         gradient_accumulation_steps = 4,
>         warmup_steps = 5,
>         num_train_epochs = 7, # Starte mit 7 Epochen (ergibt 7 * 3000 = 21000 Schritte)
>         max_steps = 70000, # WICHTIG: Dadurch wird num_train_epochs aktiv.
>         learning_rate = 2e-4,
>         logging_steps = 100, # Weniger Log-Ausgaben für bessere Übersicht
>         optim = "adamw_8bit",
>         weight_decay = 0.01,
>         lr_scheduler_type = "linear",
>         seed = 3407,
>         output_dir = "/content/drive/MyDrive/OrpheusTTS/output_august2025/",
>         report_to = "wandb", # Ändere dies, wenn du z.B. WandB nutzen möchtest
>         # Fügen Optional weitere Argumente hinzu:
>         save_steps = 1000, # Speichere Checkpoints alle 5000 Schritte
>         # save_total_limit = 2, # Halte nur die letzten 2 Checkpoints
>         # evaluation_strategy = "epoch", # Wenn du ein validation_dataset hast
>     ),
> )
> ```
> 
> And here are some audio samples: **38k Checkpoint** _"Ich frage mich wirklich, wieso einige Beispiele sehr gut und andere wiederum überhaupt nicht funktionieren. Wie kann das nur sein?"_ 🔈 https://github.com/user-attachments/assets/f88e9bf9-68ee-4e37-9259-62340b562fc8
> 
> _"Dieses Beispiel wurde mit der künstlichen, kostenlosen und ohne Internet funktionierenden Thorsten Stimme erzeugt."_ 🔈 https://github.com/user-attachments/assets/e444d6ba-1925-4f38-ad82-29a807d06f96
> 
> _"Für mich sind alle Menschen gleich, unabhängig von Geschlecht, sexueller Orientierung, Religion, Hautfarbe oder Geokoordination der Geburt."_ (i know about the funny typo 🤣) 🔈 https://github.com/user-attachments/assets/90acaf83-942c-4435-b136-b658f2acb5f1
> 
> **46k Checkpoint** _"Ich frage mich wirklich, wieso einige Beispiele sehr gut und andere wiederum überhaupt nicht funktionieren. Wie kann das nur sein?"_ 🔈 https://github.com/user-attachments/assets/4242125f-c650-405e-a16d-3b00cfe50736
> 
> _"Dieses Beispiel wurde mit der künstlichen, kostenlosen und ohne Internet funktionierenden Thorsten Stimme erzeugt."_ 🔈 https://github.com/user-attachments/assets/d65f51e8-1cad-4103-a80b-1d312077fc11
> 
> _"Irgendwie traue ich dem Frieden nicht, das wirkt manchmal wirklich sehr zufällig in Punkto Qualität."_ 🔈 https://github.com/user-attachments/assets/95e62214-a5a3-47cc-93b8-b6a4bf2f4a07
> 
> Do you have any suggestions? Is more training required or might it be some struggle on missing phonemes in training data?
> 
> Best Thorsten

The model does seem to have overfit - based on the loss curve it seems like you have finetuned it for 3 epochs - I would suggest doing fpft for 1 epoch rather than lora the script is in the repo. Happy to discuss more.

EDIT: not sure why I assumed lora - but either way 1 epoch is better. Also if you do fpft - the learning rate is really high - aim for around 5e-5=> 5e-6 depending on effective batch size. Aim for atleast EBS>4 using gradient accumulation steps if hardware constrains.

**Comment by thorstenMueller** [comment]
Thanks for your quick response. If i understand you correct it doesn't make sense to continue training in it's current configuration. I did not actively trained 3 epochs. But i needed to continue training multiple times due to automatic disconnect of google colab notebook. Now I have two ideas in mind, which one would make more sense (in your opinion).

1) Do some inference testing with earlier checkpoints (i have checkpoints from each 1000 steps) to get an impression of when the model performed better.

2) Start a new finetune from "scratch" (using unsloth?) with an adjusted training config on epoch/steps?
